packages/staking/src/features/Drawer/TransactionFail.tsx

```diff
@@ -57,23 +60,37 @@ export const TransactionFailFooter = ({ popupView }: TransactionFailProps): Reac
     portfolioMutators.executeCommand({ type: 'CancelDrawer' });
   };
 
+  const isInMemory = walletType === WalletType.InMemory;
+
   // TODO unify
   const signAndSubmitTransaction = useCallback(async () => {
     if (!delegationTxBuilder) throw new Error('Unable to submit transaction. The delegationTxBuilder not available');
+
+    if (!isInMemory) {
+      const isSupported = await isMultidelegationSupportedByDevice(walletType);
+      if (!isSupported) {
+        throw new Error('MULTIDELEGATION_NOT_SUPPORTED');
+      }
+    }
     const signedTx = await delegationTxBuilder.build().sign();
     await inMemoryWallet.submitTx(signedTx);
-  }, [delegationTxBuilder, inMemoryWallet]);
+  }, [delegationTxBuilder, inMemoryWallet, isInMemory, isMultidelegationSupportedByDevice, walletType]);
 
   const onSubmit = async () => {
     setIsLoading(true);
+
     try {
       await signAndSubmitTransaction();
       setIsLoading(false);
       portfolioMutators.executeCommand({ type: 'DrawerContinue' });
       removePassword();
-    } catch (error) {
+    } catch (error: any) {
```

Comment:
This might be better and removes the warning about using `any`
```suggestion
} catch (error: unknown) {
```

Reply:
or maybe even using the `Error` type works?
```suggestion
} catch (error: Error) {
```

Reply:
It cant be unknown becuase we need to access a fields:

```
 if (error.message === 'MULTIDELEGATION_NOT_SUPPORTED') 
```

and it cant be Error, because I get this error:

```
TS1196: Catch clause variable type annotation must be any or unknown if specified
```

Reply:
I think then you can use `unknown` and have to check for the `message` field to exist:

```ts
const isErrorType = typeof error === 'object' && error !== null && 'message' in error;

if (isErrorType && error.message === 'MULTIDELEGATION_NOT_SUPPORTED') {
  portfolioMutators.executeCommand({ type: 'HwSkipToDeviceFailure' });
}
```

Reply:
@AngelCastilloB so in many places we are actually throwing specific instances of error classes which then can be checked against `instanceof` like here:
https://github.com/input-output-hk/lace/blob/0d1d68507430eaf3a75e6fabbea58f7a04abbbdb/apps/browser-extension-wallet/src/features/dapp/components/SignData.tsx#L32

Reply:
In the SDK there are helper functions to construct these … but tbh i haven't looked deeply into why this is accepted from TS perspective while "normal" errors are not. @mkazlauskas probably knows more details about it
https://github.com/input-output-hk/cardano-js-sdk/blob/6b9fef16ffc854ee6ea8954714c94ccb2b788a63/packages/key-management/src/errors/AuthenticationError.ts#L8

Reply:
@AngelCastilloB ah and i can see that there is a workaround to check against normal errors like in your example:
https://stackoverflow.com/questions/64452484/how-can-i-safely-access-caught-error-properties-in-typescript

(the first answer points to TS not allowing `Error` directly in the catch clause but you can use `unknown` and then use the instanceof operator like normal:

```ts
} catch (e: unknown) {
    if (e instanceof Error) {
        console.log(e.message)
    }
}
```

Reply:
this does work, thanks!. Testing before pushing the fixup